clustered/client/src/test/java/org/ehcache/clustered/client/internal/service/ConnectionClosedTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.client.internal.service;
+
+import org.ehcache.Cache;
+import org.ehcache.PersistentCacheManager;
+import org.ehcache.clustered.client.config.builders.TimeoutsBuilder;
+import org.ehcache.clustered.client.internal.UnitTestConnectionService;
+import org.ehcache.config.builders.CacheManagerBuilder;
+import org.ehcache.config.builders.ResourcePoolsBuilder;
+import org.ehcache.config.units.EntryUnit;
+import org.ehcache.config.units.MemoryUnit;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+import org.terracotta.connection.Connection;
+import org.terracotta.connection.ConnectionPropertyNames;
+import org.terracotta.exception.ConnectionClosedException;
+import org.terracotta.lease.connection.LeasedConnection;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Properties;
+
+import static org.ehcache.clustered.client.config.builders.ClusteredResourcePoolBuilder.clusteredDedicated;
+import static org.ehcache.clustered.client.config.builders.ClusteringServiceConfigurationBuilder.cluster;
+import static org.ehcache.config.builders.CacheConfigurationBuilder.newCacheConfigurationBuilder;
+import static org.ehcache.config.builders.CacheManagerBuilder.newCacheManagerBuilder;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+@RunWith(Parameterized.class)
+public class ConnectionClosedTest {
+
+  private static final URI CLUSTER_URI = URI.create("terracotta://connection.com:9540/timeout");
+
+  @Parameters
+  public static ResourcePoolsBuilder[] data() {
+    return new ResourcePoolsBuilder[]{ResourcePoolsBuilder.newResourcePoolsBuilder()
```

Comment:
why parameterized since you have only one parameter?

Reply:
No it has parameters, two types of resource pools one with local tier, one without it

Reply:
Just got it.... Can you please indent like
```
  public static ResourcePoolsBuilder[] data() {
    return new ResourcePoolsBuilder[]{
      ResourcePoolsBuilder.newResourcePoolsBuilder()
        .with(clusteredDedicated("primary-server-resource", 2, MemoryUnit.MB)),
      ResourcePoolsBuilder.newResourcePoolsBuilder()
        .heap(10, EntryUnit.ENTRIES)
        .with(clusteredDedicated("primary-server-resource", 2, MemoryUnit.MB))
    };
  }
```